python/tk_framework_desktopserver/server_protocol.py

```diff
@@ -140,6 +155,10 @@ def onMessage(self, payload, is_binary):
 
         self._protocol_version = message["protocol_version"]
 
+        if message["command"]["name"] == "activate_encryption":
```

Comment:
An attacker could capture the activate_encrption message and never let it be forwarded to our server, which means the server would keep serving data as unencrypted, which needs to be supported for backwards compatibility with previous versions of Shotgun.

To close this hole, we need to make encryption mandatory for the server to handle any other requests than get_protocol_version.